File: LeetCode/30-day-leetcoding-challenge/week-4/jump-game.py
from typing import List

class Solution:
    def canJump(self, nums: List[int]) -> bool:
        pass

    def canJump_recurr(self, nums:List[int], l: int, h: int) -> bool:
        n = len(nums)
        if l > h or l > n or h > n:
            return False
        if l == h:
            return True
        if arr[l] <= 0:
            return False

        for x in range(l)

File: LeetCode/top-interview-questions-medium/dp/jump-game.cpp

#include<bits/stdc++.h>
using namespace std;
#define endl '\n'
#define INF LLONG_MAX/4
#define MOD 1e9+7
#define DEBUG(x) cerr << #x << " is " << (x) << endl;

class Solution {
public:
    bool canJump(vector<int>& nums) {
        int n = nums.size();
        bool dp[n]; memset(dp, false, sizeof(dp));
        dp[0]=true;
        for(int i=1;i<n;i++) {
            for(int j=0;j<i;j++) {
                if(dp[j] && j+nums[j]>=i) {
                    dp[i]=true;
                    break;
                }
            }
        }
        return dp[n-1];
    }
};


